docs/understand/kv-in-a-nutshell.md

```diff
@@ -0,0 +1,51 @@
+---
+layout: default
+title: KV in a Nutshell
+parent: Understanding lakeFS
+description: Brief introduction to lakeFS over KV
+nav_order: 25
+has_children: false
+---
+# lakeFS with Key Value Store
+{: .no_toc }
+
+{% include toc.html %}
+
+Starting at version 0.80.0, lakeFS abandoned the tight coupling to [PostgreSQL](https://en.wikipedia.org/wiki/PostgreSQL) and moved all database operations to work over [Key-Value Store](https://en.wikipedia.org/wiki/Key%E2%80%93value_database)
+
+While SQL databases, and Postgres among them, have their obvious advantages, we felt that the tight coupling to Postgres is limiting our users and so, lakeFS with Key Value Store is introduced.
+Our KV Store implements a generic interface, with methods for `Get`, `Set`, `Compare-and-Set`, `Delete` and `Scan`. Each entry is represented by a [`partition`, `key`, `value`] triplet. All these fields are generic byte-array, and the using module has maximal flexibility on the format to use for each field
+
+Under the hood, our KV implementation relies on a backing DB, which persists the data. Theoretically, it could be any type of database and out of the box, we already implemented drivers for [DynamoDB](https://en.wikipedia.org/wiki/Amazon_DynamoDB), for AWS users, and [PostgreSQL](https://en.wikipedia.org/wiki/PostgreSQL), using its relational nature to store a KV Store. More databases will be supported in the future, and lakeFS users and contributors can develop their own driver to use their own favorite database. For experimenting purposes, an in-memory KV store can be used, though it obviously lack the persistency aspect
+
+In order to store ref store objects (that is `Repositories`, `Branches`, `Commits`, `Tags`, and `Uncommitted Objects`), lakeFS implements another layer over the generic KV Store, which supports serialization and deserialization of these objects as [protobuf](https://en.wikipedia.org/wiki/Protocol_Buffers). As this layer relies on the generic interface of the KV Store layer, it is totally agnostic to whichever store implementation is in use, gaining our users the maximal flexibility
+
+For further reading, please refer to our [KV Design](https://github.com/treeverse/lakeFS/blob/master/design/accepted/metadata_kv/index.md)
+
+## Optimistic Locking with KV
+
+One important key difference between SQL databases and Key Value Store is the ability to lock resources. While this is a common practice with relational databases, Key Value stores lack this ability. Let us take a look at a common lakeFS flow, `Commit`, during which several ref store operations are performed:
```

Comment:
They don't lack the ability but it is a less common practice I think

Reply:
I believe KV Stores (at least DynamoDB) are connectionless, and as such cannot support locking (I think a lock should be bound to a connection, otherwise , in case of disconnection, no one will be able to unlock it)

Reply:
I think we shouldn't state something as a fact: "While this is a common practice with relational databases, Key Value stores lack this ability" if we are uncertain of it

Reply:
You convinced me. Fixed